src/services/blocks/BlocksService.ts

```diff
@@ -500,14 +536,29 @@ export class BlocksService extends AbstractService {
 				version.specVersion.toNumber(),
 			];
 
-			runtimeDoesNotMatch =
-				specName !== api.runtimeVersion.specName.toString() ||
-				specVersion !== api.runtimeVersion.specVersion.toNumber();
+			const hardcodedChainIncluded = hardcodedChains.includes(specName);
 
-			if (runtimeDoesNotMatch) {
+			if (!hardcodedChainIncluded) {
```

Comment:
This isn't the check we want to be doing.

1) We only support polkadot based chains + dock for fee calc as of right now
2) We need to get and expand metadata whenever there is a new runtime for one of the chains we support, i.e a live runtime upgrade and we have not hardcoded the values

This being said this whole branch of logic can probably be removed. I think the simplest way to do this is lazy filling of the cache thing. So basically somewhere before we need any weights we do the equivalent of the following logic
```javascript
metaConstsCache[runtimeVersion] ||= this.getDecorateAndExtractWeight(blockNumber)
```

And then every time forward we need weights for the runtime version its already in the `metaConstsCache`, so we don't need al this extra logic.